src/main/java/org/opentripplanner/api/mapping/LegMapper.java

```diff
@@ -100,12 +110,90 @@ else if (domain.pathway) {
         );
         api.boardRule = domain.boardRule;
         api.alightRule = domain.alightRule;
+
+        api.pickupBookingInfo = toApi(domain.pickupBookingInfo);
+        api.dropOffBookingInfo = toApi(domain.dropOffBookingInfo);
+
         api.rentedBike = domain.rentedVehicle;
         api.walkingBike = domain.walkingBike;
 
         return api;
     }
 
+    private ApiBookingInfo toApi(BookingInfo info) {
+        if (info != null) {
+
+            return new ApiBookingInfo(
+                    toApi(info.getContactInfo()),
+                    toApi(info.bookingMethods()),
+                    toApi(info.getEarliestBookingTime()),
+                    toApi(info.getLatestBookingTime()),
+                    info.getMinimumBookingNotice(),
+                    info.getMaximumBookingNotice(),
+                    info.getMessage(),
+                    info.getPickupMessage(),
+                    info.getDropOffMessage()
+            );
+        }
+        else {
+            return null;
+        }
+    }
+
+    private ApiContactInfo toApi(ContactInfo info) {
+        if (info != null) {
+            return new ApiContactInfo(
+                    info.getContactPerson(),
+                    info.getPhoneNumber(),
+                    info.geteMail(),
+                    info.getFaxNumber(),
+                    info.getInfoUrl(),
+                    info.getBookingUrl(),
+                    info.getAdditionalDetails()
+            );
+        }
+        else {
+            return null;
+        }
+    }
+
+    private EnumSet<ApiBookingMethod> toApi(EnumSet<BookingMethod> m) {
+           if(m !=null) {
+
+        return m.stream()
+                .map(this::toApi)
+                .collect(Collectors.toCollection(() -> EnumSet.noneOf(ApiBookingMethod.class)));
+        } else {
+            return null;
+        }
+    }
+
+    private ApiBookingTime toApi(BookingTime time) {
+        if(time != null) {
+            return new ApiBookingTime(time.getTime().toSecondOfDay(), time.getDaysPrior());
+        } else {
+            return null;
+        }
+    }
+
+    private ApiBookingMethod toApi(BookingMethod m) {
+        switch (m) {
```

Comment:
These can be represented just by a string

Reply:
Thanks. I've simplified the API model.

Reply:
We should still list all values here, as now changing the internal model will change the API